code/script.js

```diff
@@ -229,97 +231,191 @@ const setSecret = () => {
+
+    if (category === 'accessories') {
+      if (secret.accessories.includes(value)) {
+        filterCharacters(true)
+      }
+      else {
+        filterCharacters(false)
+      }
+    }
+    else {
+      if (secret.abilities.includes(value)) {
+        filterCharacters(true)
+      }
+      else {
+        filterCharacters(false)
+      }
+    }
   }
 }
 
-// It'll filter the characters array and redraw the game board.
+// It will filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
   const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
+  let stringPropertyName = '' 
+  let arrayPropertyName = '' 
+
+
+
+
+// Show the correct alert message for different categories
+  if (category === 'hair color') {
+    stringPropertyName = 'hairColor';
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the Avenger have ${value} hair! Keep all Avengers that have ${value} hair`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the Avenger doesn't have ${value} hair! Remove all Avengers that have ${value} hair`
       )
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  }
+  else if (category === 'what species') {
+    stringPropertyName = 'whatSpecies'
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, the Avenger is a ${value}! Keep all Avengers that is a ${value}`
+      )
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `No, the Avenger isn´t a ${value}! Remove all Avenger that isn´t a ${value}`
+      )
+    }
+  } else if (category === 'accessories') {
+    arrayPropertyName = 'accessories';
+    if (keep) {
+      alert(
+        `Yes, the Avenger have a ${value}! Keep all Avengers that have a ${value}`
+      )
+    } else {
+      alert(
+        `No, the Avenger doesn't have a ${value}! Remove all Avengers that have a ${value}`
+      )
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+  else if (category === 'abilities') {
```

Comment:
So happy that my simplifying of the code helped you. I can say that with out writing out everything in this way I hadn't understand one bit of this part of the code.

I think in the beginning of our journey to become a front end developer this is the way to do it. Just to understand what each bit of code are doing.

Reply:
So thankful for this, made my week!! And I totally agree.